File: main/backend/src/Ecommerce.BLL/Services/InventoryService.cs
using Ecommerce.BLL.Interfaces;
using Ecommerce.Common.DTOs;
using Ecommerce.Common.Enums;
using Ecommerce.Common.Exceptions;
using Ecommerce.DAL.Interfaces;

namespace Ecommerce.BLL.Services;

public class InventoryService : IInventoryService
{
    private readonly IUnitOfWork _unitOfWork;

    public InventoryService(IUnitOfWork unitOfWork)
    {
        _unitOfWork = unitOfWork;
    }

    public async Task<InventorySummaryDto> GetInventorySummaryAsync()
    {
        var allProducts = await _unitOfWork.Products.GetAllAsync();
        var products = allProducts.Where(p => p.IsActive).ToList();

        var totalSku = products.Count;
        var totalStockItems = products.Sum(p => p.StockQuantity);
        var totalStockValue = products.Sum(p => p.BasePrice * p.StockQuantity);
        var lowStockCount = products.Count(p => p.StockQuantity <= 10);

        var items = products.Select(p => new InventoryItemDto
        {
            Id = p.Id,
            Name = p.Name,
            Sku = p.Sku,
            CategoryName = p.Category?.Name ?? "Fashion",
            ImageUrl = p.ImageUrl,
            Price = p.BasePrice,
            StockQuantity = p.StockQuantity,
            StockStatus = p.StockQuantity == 0 ? "OUT_OF_STOCK" : (p.StockQuantity <= 10 ? "LOW_STOCK" : "IN_STOCK"),
            SafetyThreshold = 10
        }).ToList();

        return new InventorySummaryDto
        {
            TotalSku = totalSku,
            TotalStockItems = totalStockItems,
            TotalStockValue = totalStockValue,
            LowStockAlertCount = lowStockCount,
            Items = items
        };
    }

    public async Task<bool> RestockProductAsync(RestockRequestDto request)
    {
        var product = await _unitOfWork.Products.GetByIdAsync(request.ProductId);
        if (product == null)
            throw new NotFoundException("Sản phẩm", request.ProductId);

        product.StockQuantity += request.AdditionalQuantity;
        if (product.StockQuantity > 10)
            product.StockStatus = StockStatus.InStock;
        else if (product.StockQuantity > 0)
            product.StockStatus = StockStatus.LowStock;

        _unitOfWork.Products.Update(product);
        await _unitOfWork.SaveChangesAsync();
        return true;
    }
}

File: frontend/src/services/inventory.service.ts
import apiClient from './api-client';
import type { InventorySummary, RestockRequest } from '@/types';
import { fallbackProducts } from '@/lib/mock-data';

export const inventoryService = {
  async getInventory(): Promise<InventorySummary> {
    try {
      const res = await apiClient.get('/inventory');
      return res.data;
    } catch {
      return {
        totalSku: fallbackProducts.length,
        totalStockItems: fallbackProducts.reduce((acc, p) => acc + p.stockQuantity, 0),
        totalStockValue: fallbackProducts.reduce((acc, p) => acc + p.basePrice * p.stockQuantity, 0),
        lowStockAlertCount: fallbackProducts.filter(p => p.stockQuantity <= 10).length,
        items: fallbackProducts.map(p => ({
          id: p.id,
          name: p.name,
          sku: p.sku,
          categoryName: p.categoryName || 'Fashion',
          imageUrl: p.imageUrl,
          price: p.basePrice,
          stockQuantity: p.stockQuantity,
          stockStatus: p.stockQuantity === 0 ? 'OUT_OF_STOCK' : (p.stockQuantity <= 10 ? 'LOW_STOCK' : 'IN_STOCK'),
          safetyThreshold: 10,
        })),
      };
    }
  },

  async restock(data: RestockRequest): Promise<boolean> {
    try {
      await apiClient.post('/inventory/restock', data);
      return true;
    } catch {
      const p = fallbackProducts.find(x => x.id === data.productId);
      if (p) p.stockQuantity += data.additionalQuantity;
      return true;
    }
  },
};
